storage/jellyfish_merkle/src/node_type/mod.rs

```diff
@@ -265,14 +271,21 @@ impl From<LeafNode> for Node {
 impl InternalNode {
     /// Creates a new Internal node.
     pub fn new(children: Children) -> Self {
+        // Assert the internal node must have 1+ children. If only has 1 chlid, it cannot be leaf
```

Comment:
This comment is confusing.  Please mention why (because otherwise this child should have been stored as a leaf in this node's parent).  Also "If it only has one child, it cannot be a leaf node".